tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc

```diff
@@ -625,7 +626,35 @@ StatusOr<std::unique_ptr<Executable>> NVPTXCompiler::RunBackend(
   }
 
   string ptx;
-  {
+
+  // Generate the PTX or load it if provided.
+  // If the xla_gpu_ptx_code options is set, be explicit when a file is used
+  // and warn when a file is not used to ease catching typo in filename.
+  string prefix = FilenameFor(*module, ptx);
+  string ptx_filename;
+  for (const string filename : module->config().debug_options().xla_gpu_ptx_code()) {
+    // To ease comparing many PTX versions, accept different suffix then
+    // the original filename.
+    if(absl::StartsWith(filename, prefix)) {
+      ptx_filename = filename;
+      VLOG(0) << "RunBackend() - Will load PTX from file: " << filename;
+      break;
+    } else {
+      VLOG(0) << "RunBackend() - For module with prefix '" << prefix
+              << "', we skip PTX code file: " << filename;
+    }
+  }
+  if (module->config().debug_options().xla_gpu_ptx_code().size() > 0 &&
+      ptx_filename.size() == 0) {
+    VLOG(0) << "RunBackend() - For module with prefix '" << prefix
+            << "', we did not found a PTX file to load.";
```

Comment:
This VLOG would triggers for the same cases as the one on line 643, right? Is either one good enough?